scss/components/_forms.scss

```diff
@@ -72,3 +72,10 @@
 		display: block;
 	}
 }
+
+.newsletter-form {
```

Comment:
you created a component for it but then you put the code in another component. We don't need to create neither a component nor a compounds. to get that behaviour there is a specific bootstrap element ready to use, it's called Input group https://getbootstrap.com/docs/4.0/components/input-group/